offchain/danaswap-browser/src/DanaSwap/Browser/Home.purs

```diff
@@ -19,18 +24,239 @@ main = HA.runHalogenAff do
     Just elem -> void $ runUI component unit elem
 
+type State =
+  { pools :: Array Pool
+  , currentPool :: Maybe Pool
+  , activity :: Activity
+  }
+
+data Action
+  = SetCurrentPool Pool
+  | ResetCurrentPool
+  | SetActivity Activity
+
+data Activity
+  = Swap
+  | AddLiquidity
+  | WithdrawLiquidity
+
+derive instance eqActivity :: Eq Activity
+
 component
   :: forall q i o m
    . MonadAff m
   => H.Component q i o m
 component =
   H.mkComponent
-    { initialState: const unit
+    { initialState: const
+        { pools:
+            [ Pool { assetA: Asset { name: "ADA", value: fromInt 1200 }, assetB: Asset { name: "DANA", value: fromInt 1000 }, fee: 4 }
+            , Pool { assetA: Asset { name: "ADA", value: fromInt 10000 }, assetB: Asset { name: "USDC", value: fromInt 4000 }, fee: 4 }
+            , Pool { assetA: Asset { name: "ADA", value: fromInt 120 }, assetB: Asset { name: "HOSKY", value: fromMaybe (fromInt 0) (fromString "3000000100") }, fee: 4 }
+            , Pool { assetA: Asset { name: "DADA", value: fromInt 3000 }, assetB: Asset { name: "USDC", value: fromInt 1000 }, fee: 4 }
+            ]
+        , currentPool: Nothing
+        , activity: Swap
+        }
     , render
     , eval: H.mkEval $ H.defaultEval
+        { handleAction = handleAction
+        }
     }
   where
-  render _ =
-    HH.div_
-      [ HH.text "this is Home"
+  handleAction :: forall slots. Action -> H.HalogenM State Action slots o m Unit
+  handleAction = case _ of
+    ResetCurrentPool -> H.modify_ _ { currentPool = Nothing }
+    SetCurrentPool pool -> H.modify_ _ { currentPool = Just pool }
+    SetActivity activity -> H.modify_ _ { activity = activity }
+
+  render :: forall slots. State -> H.ComponentHTML Action slots m
+  render { pools, currentPool, activity } =
+    HH.section
+      [ mkClass "section my-6 pb-6" ]
+      $
+        [ HH.div
+            [ mkClass "columns is-mobile" ]
+            [ HH.div [ mkClass "pools-table-header column" ]
+                [ HH.div [ mkClass $ "columns" ]
+                    [ HH.div [ mkClass "column has-text-left" ] [ HH.text "Pool" ]
+                    ]
+                ]
+            , HH.div [ mkClass "pools-table-header column" ]
+                [ HH.div [ mkClass $ "columns" ]
+                    [ HH.div [ mkClass "column has-text-centered" ] [ HH.text "Value" ]
+                    ]
+                ]
+            , HH.div [ mkClass "pools-table-header column" ]
+                [ HH.div [ mkClass $ "columns" ]
+                    [ HH.div [ mkClass "column has-text-centered" ] [ HH.text "Fee" ]
+                    ]
+                ]
+            , HH.div [ mkClass "column" ] [ HH.text " " ]
+            ]
+
+        ]
+      <> concat
+        ( renderPool
+            <$> pools
+        )
+    where
+    renderPool pool@(Pool p) =
+      [ HH.div
+          [ mkClass "pools-table-row columns is-mobile is-vcentered p-3 my-3"
```

Comment:
Should this be `role=button` or something that indicates that this is clickable? What about `tabindex` and `:focus`? Are we missing accessibility?

Reply:
good point, will add `role` to clickable buttons, for `tabindex` and `:focus`, I reckon it might be overkill for the MVP

Reply:
like combobox, for testing and such sure, but I don't think it should go to production without keyboard support.

Reply:
This can be shelved to post prototyping/MVP/alpha, but should be considered a bug for even a beta.